crates/syntax/src/ast/edit_in_place.rs

```diff
@@ -54,6 +55,80 @@ impl GenericParamsOwnerEdit for ast::Fn {
     }
 }
 
+impl ast::Fn {
+    /// Adds a new generic param to the function using `SyntaxEditor`
+    pub fn syntax_editor_add_generic_param(
```

Comment:
Thank you for the review

> but I do think `edit_in_place.rs` is probably the wrong place.

I completely agree. I actually mentioned this earlier here: https://github.com/rust-lang/rust-analyzer/pull/18483#discussion_r1838496721

I got a bit carried away focusing on getting the assist to work and forgot about this.

So, I’ve now moved it to a method of `SyntaxEditor`, but I created a new file for these edits, similar to `crates/syntax/src/ast/edit_in_place.rs`. I thought this might be a better place since `crates/syntax/src/syntax_editor.rs` mainly contains editing operations like `new`, `merge`, `insert`, `replace`, `delete`, etc.

But let me know if you’d prefer the method to be in `crates/syntax/src/syntax_editor.rs`, and I can update the code.

Reply:
That seems fine to me! We can always move things around later.